google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractResultSet.java

```diff
@@ -540,88 +539,25 @@ private static Struct decodeStructValue(Type structType, ListValue structValue)
 
     static Object decodeArrayValue(Type elementType, ListValue listValue) {
       switch (elementType.getCode()) {
-        case BOOL:
-          // Use a view: element conversion is virtually free.
-          return Lists.transform(
-              listValue.getValuesList(),
-              input -> input.getKindCase() == KindCase.NULL_VALUE ? null : input.getBoolValue());
         case INT64:
           // For int64/float64 types, use custom containers.  These avoid wrapper object
           // creation for non-null arrays.
           return new Int64Array(listValue);
         case FLOAT64:
           return new Float64Array(listValue);
+        case BOOL:
```

Comment:
nit: I think it would be better to keep BOOL conversion as is. That conversion also does not really convert the array, but creates a 'view' of the current list and only looks at the value if someone tries to actually get one of the values in the list.